test/core/address_utils/sockaddr_utils_fuzzer_test.cc

```diff
@@ -18,28 +18,32 @@
 #include <string.h>
 
 #include <string>
+#include <vector>
 
 #include "absl/log/check.h"
 #include "absl/status/statusor.h"
+#include "fuzztest/fuzztest.h"
 #include "src/core/lib/address_utils/sockaddr_utils.h"
 #include "src/core/lib/iomgr/resolve_address.h"
 #include "src/core/lib/iomgr/resolved_address.h"
 #include "src/core/util/uri.h"
 
-bool squelch = true;
+using fuzztest::Arbitrary;
+using fuzztest::VectorOf;
 
-extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
-  if (size > GRPC_MAX_SOCKADDR_SIZE) return 0;
+void CheckUriIsParseable(std::vector<uint8_t> buffer) {
   grpc_resolved_address address;
   memset(&address, 0, sizeof(address));
-  memcpy(address.addr, data, size);
-  address.len = size;
-
+  if (!buffer.empty()) {
```

Comment:
Is there any value in continuing here if the buffer is empty? We could do an early return.

Reply:
main thing is to avoid UB with memcpy from a nullptr buffer (like we get with zero length)
constrained the domain instead - pretty sure it's equivalent and avoids the test branch